p2panda-sync/src/log_sync.rs

```diff
@@ -39,20 +41,21 @@ where
 static LOG_SYNC_PROTOCOL_NAME: &str = "p2panda/log_sync";
 
 #[derive(Clone, Debug)]
-pub struct LogSyncProtocol<S, T, E> {
-    pub topic_map: S,
-    pub store: MemoryStore<T, E>,
+pub struct LogSyncProtocol<TM, L, E> {
+    pub topic_map: TM,
+    pub store: MemoryStore<L, E>,
 }
 
 #[async_trait]
-impl<'a, S, T, E> SyncProtocol<'a> for LogSyncProtocol<S, T, E>
+impl<'a, T, TM, L, E> SyncProtocol<T, 'a> for LogSyncProtocol<TM, L, E>
 where
-    S: Debug + TopicMap<TopicId, T> + Send + Sync,
-    T: Clone
+    T: Clone + Debug + Send + Sync + Serialize + for<'de> Deserialize<'de>,
```

Comment:
I think this should be the newly introduced topic trait here?

Reply:
The `Topic` trait is defined in `p2panda-net` and primarily used for guaranteeing that a `[u8; 32]` id is available for gossip overlay identification. This id isn't required in `p2panda-sync` for anything, so it might be strange to add this trait as a bound here.

Reply:
Thanks for clarifying, I think this doesn't fully get across looking at the APIs, both concepts are called topic right now (for gossip and sync) and the trait bounds do not communicate themselves well (I think we need to combine the concepts or introduce a second super trait).

Ill write more thoughts in a comment with some ideas!